testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TransactionCheckerSingleton.java

```diff
@@ -40,8 +40,8 @@
 @Remote
 @ManagedBean
 public class TransactionCheckerSingleton implements TransactionCheckerSingletonRemote {
-    private int committed, prepared, rolledback;
-    private int  synchronizedBegin, synchronizedBefore, synchronizedAfter,
+    private volatile int committed, prepared, rolledback;
+    private volatile int synchronizedBegin, synchronizedBefore, synchronizedAfter,
         synchronizedAfterCommitted, synchronizedAfterRolledBack;
     private Collection<String> messages = new ArrayList<>();
```

Comment:
I suspect this might need to use a concurrent collection (or at least synchronized wrapper).

Reply:
See above comment.